drivers/libusb0.c

```diff
@@ -489,11 +472,10 @@ static int libusb_open(usb_dev_handle **udevp,
 			/* this method requires at least libusb 0.1.8:
 			 * it force device claiming by unbinding
 			 * attached driver... From libhid */
-			retries = MAX_RETRY;
 #ifdef WIN32
 			usb_set_configuration(udev, 1);
 #endif
-
+			retries = 3;
```

Comment:
MAX_RETRY was introduced in the reverted commit and made sense there because it was used in many places. Now it is only used in one place, and I thought "retries = 3" is as readable as packing it into a macro that is defined somewhere else in the code only to be used once. In any case I think it should be commented that 3 is just an arbitrary number and guesswork, not based on experience or logic.

Technically "3" is the number of tries, not retries. Would you like me to reintroduce MAX_RETRY and make it loop 1 + MAX_RETRY?

Finally, are we sure the retries for string descriptors are actually needed? #414 seems unclear on this, I only understood that the HID descriptors needed retries, not sure about any string descriptors. In general we shouldn't add extra code "just in case, cannot hurt" unless it is has been proven to help at least in one real case.

Reply:
Ok, makes sense. For "being sure" - I can only suggest scrolling back in `git blame` (or GH UI equivalent) likely on `libusb0.c` and check what comments might appear in commit or PR that introduced those?..

Reply:
Oh, in the retries code (if we want to keep it :) ) I added a delay, that was not there before. This is also guesswork "can probably help". But I meant to use 50 ms, not us.

Some arguments for not doing retries are that USB does low-level retries already, and that e.g. any issues with concurrent access from other code or OS should be dealt with properly instead of papering over with retries.

Reply:
I did try to look up the reason for the retries, commit 0fa5687a took me to #414 as mentioned in my commit message.

BTW, although you are attributed as author, with kudos to Sam Varshavchik, it was actually @aquette that came up with the code in https://github.com/networkupstools/nut/issues/414#issuecomment-292143944.

Looking at log1.txt and log2.txt from https://github.com/networkupstools/nut/issues/414#issuecomment-291639104 it actually looks like there is a spurious Product string retrieval failure, that was not seen again after applying @aquette's patch. Sure that patch includes another fix as well, but I agree that looks unrelated. Sadly we don't see "retrying..." in the logs, so we cannot for sure know if the patch actually helped.

So I'd say we keep the retries, comment that this is introduced for "Tripp Lite", make sure we log any retries for future evaluation after it has been tested with "Tripp Lite" products for a while.

Reply:
I see some other code is using MAX_TRIES and I think that is better than 1 + MAX_RETRY, unless you prefer something more specific like MAX_STRING_DESC_TRIES.

Reply:
I wonder if those devices had a sort of throttling when overwhelmed with queries, so sleeping helped? A comment makes sense, and a domain-specific macro name too.

Reply:
There was no sleeping in the original retries, so it is apparently not needed. I just think it can help (and cannot hurt(TM)), for instance if used with faster computers than originally tested on.